src/Themes.tsx

```diff
@@ -1,75 +1,160 @@
 import {List} from "immutable";
 import * as React from "react";
-import {branch, compose, lifecycle, renderNothing, withHandlers, withState} from "recompose";
+import {STORY_RENDERED} from "@storybook/core-events";
+import {branch, compose, lifecycle, mapProps, renderNothing, withHandlers, withState} from "recompose";
 import {Theme} from "./types/Theme";
+import {metadataKey} from "./constants";
 
 export interface ThemeProps {
     channel: any;
     api: any;
     active: boolean;
 }
 
+export interface ThemeMetadataParams {
+    themes?: Theme[];
+    singleThemeMessage?: string;
+    showSingleThemeButton?: boolean;
+    buttonAttributes?: string[];
+}
+
+interface ThemeExtraProps {
+    singleThemeMessage?: string;
+    showSingleThemeButton: boolean;
+    buttonAttributes: string[];
+}
+
 interface ThemeState {
-    theme: Theme;
-    setTheme: (theme: Theme) => void;
-    themes: List<Theme>;
-    setThemes: (themes: List<Theme>) => void;
+    stateTheme: Theme;
+    setStateTheme: (theme: Theme) => void;
+    stateThemes: List<Theme>;
+    setStateThemes: (themes: List<Theme>) => void;
 }
 
 interface ThemeHandler {
     onSelectTheme: (theme: Theme) => void;
     onReceiveThemes: (theme: Theme[]) => void;
+    onStoryRender: (storyId: string) => void;
 }
 
-type BaseComponentProps = ThemeProps & ThemeState & ThemeHandler;
+type BaseComponentProps = ThemeProps & ThemeExtraProps & ThemeState & ThemeHandler;
 
-const BaseComponent: React.SFC<BaseComponentProps> = ({onSelectTheme, themes, theme}) => (
-    <div style={RowStyle}>
-        {themes.map((th, i) => {
-            const buttonStyle = th === theme ? SelectedButtonStyle : ButtonStyle;
-            return <div style={buttonStyle} key={i} onClick={() => onSelectTheme(th)}>{th.name}</div>;
-        }).toArray()}
-    </div>
-);
+const BaseComponent: React.SFC<BaseComponentProps> =
+    ({onSelectTheme, stateThemes, stateTheme, singleThemeMessage, showSingleThemeButton, buttonAttributes}) => (
+        <div>
+            {stateThemes.size === 1 && singleThemeMessage && (
+                <div style={MessageStyle}>{singleThemeMessage}</div>
+            )}
+            {(stateThemes.size > 1 || showSingleThemeButton) && (
+                <div style={RowStyle}>
+                    {stateThemes.map((th: Theme) => {
+                        const style: React.CSSProperties = ((th === stateTheme) ? SelectedButtonStyle : ButtonStyle);
+                        const themeNameProps: object = buttonAttributes.reduce(
+                            (obj: object, attr: string) => ({ ...obj, [attr]: th.name}), {},
```

Comment:
Is this just going to always set the value of every buttonAttribute passed to `th.name`? If so that doesn't feel particularly useful - maybe `buttonAttributes` would be better as a `Map`? (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map)

Reply:
@kyanmichael as discussed internally in Slack, the buttonAttributes are now surplus to requirements and can be removed 😄

Reply:
Indeed! Many thanks for the discussion :slightly_smiling_face: Removed in e0af2cd :thumbsup: